### PR TITLE
feat: Save MessageMedia to a file

### PR DESCRIPTION
**Pull request by purpshell**
In this pull request I add a function to MessageMedia to save files, since this has apparently became a headache, and I though having a function that does this by default is fitting. 

Example usage:
```js
const media = await msg.downloadMedia();
await media.toFilePath('/home/purpshell/Documents/message') // the code adds the extension for you (if you enter a directory we will use the filename property (if it's null then we'll throw an error))
```

### CODE REVIEW

File: src/structures/MessageMedia.js [hunk 1]
```diff
@@ -98,6 +98,32 @@ class MessageMedia {
...
+     * media.toFilePath('/home/purpshell/Documents/message') // the code adds the extension for you (if you enter a directory we will use the filename property (if it's null then we'll throw an error))
+     * @returns {null} no need to return anything
+     */
+    async toFilePath(filePath) {
+        const ext = mime.getExtension(this.mimetype);
+        if (fs.existsSync(filePath)) {
+            const stat = await fs.promises.stat(filePath);
+            if (stat.isDirectory()) {
+                if (this.filename){
+                    filePath += this.filename;
```
Comment (RakSrinaNa) [inline src/structures/MessageMedia.js:116]: Wouldn't this break if the input doesn't end with a folder separator ?

Like with input `/path/to/folder` it'll save to `/path/to/folderfilename.ext`

File: src/structures/MessageMedia.js [hunk 2]
```diff
@@ -98,6 +98,32 @@ class MessageMedia {
...
+                } else {
+                    throw Error('You passed in a directory but the filename is empty');
+                }
+            }
+        }
+        if (!filePath.includes('.')) {
```
Comment (RakSrinaNa) [inline src/structures/MessageMedia.js:122]: What if the provided input folder is `./path/to/myfile` or `/path/to.my.folder/myfile`, the extension wouldn't be added.

Comment (PurpShell) [reply src/structures/MessageMedia.js:122]: you're right, ill make some changes

File: src/structures/MessageMedia.js (hunk 1 above)
Comment (PurpShell) [reply src/structures/MessageMedia.js:116]: aight

**Review by Rakambda — APPROVED** [review]

**Review by tuyuribr — APPROVED** [review]
LGTM

**Pull request closed without merging**